release/golden_notebook_tests/workloads/torch_tune_serve_test.py

```diff
@@ -91,49 +94,37 @@ def training_loop(config):
         train_epoch(train_loader, model, criterion, optimizer)
         validation_loss = validate_epoch(validation_loader, model, criterion)
 
-        train.save_checkpoint(model_state_dict=model.module.state_dict())
-        train.report(**validation_loss)
+        session.report(
+            validation_loss,
+            checkpoint=TorchCheckpoint.from_state_dict(model.module.state_dict()),
+        )
 
 
 def train_mnist(test_mode=False, num_workers=1, use_gpu=False):
-    trainer = Trainer(backend="torch", num_workers=num_workers, use_gpu=use_gpu)
-    TorchTrainable = trainer.to_tune_trainable(training_loop)
-
-    return tune.run(
-        TorchTrainable,
-        num_samples=1,
-        config={
-            "lr": tune.grid_search([1e-4, 1e-3]),
-            "test_mode": test_mode,
-            "batch_size": 128,
+    trainer = TorchTrainer(
+        training_loop,
+        scaling_config=ScalingConfig(num_workers=num_workers, use_gpu=use_gpu),
+    )
+    tuner = Tuner(
+        trainer,
+        param_space={
+            "train_loop_config": {
+                "lr": tune.grid_search([1e-4, 1e-3]),
+                "test_mode": test_mode,
+                "batch_size": 128,
+            }
         },
-        stop={"training_iteration": 2},
-        verbose=1,
-        metric="val_loss",
-        mode="min",
-        checkpoint_at_end=True,
+        tune_config=TuneConfig(
+            metric="val_loss",
+            mode="min",
+            num_samples=1,
+        ),
+        run_config=RunConfig(
+            verbose=1,
+        ),
     )
 
-
-def get_remote_model(remote_model_checkpoint_path):
```

Comment:
do we still need get_remote_model for the client case?

Reply:
We shouldn't need it with a checkpoint object.